ledger/shelley_test.go

```diff
@@ -0,0 +1,35 @@
+package ledger
+
+import (
+	"testing"
+)
+
+func TestNonceUnmarshalCBOR(t *testing.T) {
+	testCases := []struct {
+		name string
+		data []byte
+	}{
+		{
+			name: "NonceType0",
+			data: []byte{0x81, 0x00},
+		},
+		{
+			name: "NonceType1",
+			data: []byte{0x82, 0x01, 0x01},
+		},
+		{
+			name: "UnsupportedNonceType",
+			data: []byte{0x83, 0x00, 0x01, 0x02},
+		},
```

Comment:
This doesn't actually fail. Your unmarshal function is checking the first value in the list (`0` in this case), which is a supported value. Instead I'd use something like `8102` (`[2]`), which will cause an error.